src/main/java/UseCase/createPlaylist.java

```diff
@@ -5,26 +5,41 @@
 import InputData.newPlaylistInputData;
 import Database.playlistAccessInterface;
 import Database.playlistLibrary;
+
+//import OutputData.newPlaylistOutputData;
+//import OutputBoundary.newPlaylistOutputBoundary;
+
 public class createPlaylist implements newPlaylistInputBoundary {
+    //TODO: Do I need to create an output data instance attribute ? I dont know if this is a data structure
 
+    public createPlaylist(){}
+//    private final newPlaylistOutputBoundary newPlaylistOutputBoundary;
+//    public createPlaylist(newPlaylistOutputBoundary newPlaylistOutputBoundary){
+//        this.newPlaylistOutputBoundary = newPlaylistOutputBoundary;
+//    }
     /**
-     * @param playlistInputData holds necessary data to instantiate a new playlist
+     * @param newplaylistInputData holds necessary data to instantiate a new playlist
      * if hasFirstSong() is false, then newPlaylist() will instantiate an empty playlist and vice versa if it's value
      * is true
      */
-    public String newPlaylist(newPlaylistInputData playlistInputData) {
+    public String newPlaylist(newPlaylistInputData newplaylistInputData) {
         Playlist playlist;
         playlistAccessInterface library = playlistLibrary.getInstance();
-        if (!playlistInputData.hasFirstSong()) {
-            playlist = new Playlist(playlistInputData.getId(), playlistInputData.getPlaylistName(), playlistInputData.getOwner()
-                    );
-
+        if (!newplaylistInputData.hasFirstSong()) {
+            playlist = new Playlist(newplaylistInputData.getId(), newplaylistInputData.getPlaylistName(),
+                    newplaylistInputData.getOwner());
         }
         else{
-            playlist = new Playlist(playlistInputData.getId(), playlistInputData.getPlaylistName(), playlistInputData.getOwner(),
-                     playlistInputData.getFirstSong());
+            playlist = new Playlist(newplaylistInputData.getId(), newplaylistInputData.getPlaylistName(), newplaylistInputData.getOwner(),
+                     newplaylistInputData.getFirstSong());
         }
+        newplaylistInputData.getOwner().addPlaylist(playlist);
+        // save the newly created playlist to the RegularUser's playlist list
         library.savePlaylist(new playlistDsData(playlist));
+
+        // TODO: new potential implementation of sending confirmation message via outputdata, check correctness
+//        newPlaylistOutputData outputData = new newPlaylistOutputData("Playlist created!");
+//        this.newPlaylistOutputBoundary.playlistCreationConfirmation(outputData);
         return "Playlist created!";
```

Comment:
If I'm understanding your code + clean architecture correctly -- and if I'm not, let me know! -- I don't think you need to directly return the string. You should store the OutputBoundary/Presenter as an attribute and then call its function with the outputData (i.e. the string) as a parameter.

Reply:
I understand that what i wrote might be a bit confusing! To try to make things clearer, i think what i'm envisioning is your createPlaylist stores the presenter/outputboundary as an instance attribute. then in the method, construct the outputdata, then call the presenter's function. you don't need to store the outputdata as an instance attribute itself, but you could if you want to.